### PR TITLE
Simplify mmap interception

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

**Comment by Saviq** [comment]
Building in a PPA here:

https://launchpad.net/~saviq/+archive/ubuntu/ppa/+sourcepub/15134955/+listing-archive-extra

File: tests/mir_test_framework/mmap_wrapper.cpp [hunk 1]
```diff
@@ -39,35 +39,26 @@ mtf::MmapHandlerHandle mtf::add_mmap_handler(MmapHandler handler)
...
+        *(void **)(&real_mmap) = dlsym(RTLD_NEXT, "mmap64");
+    }
 
-#if _FILE_OFFSET_BITS == 64
-    // Empirically, mmap64 is NOT defined everywhere, but on 64-bit platforms this is appropriate
+    // Empirically, mmap64 is NOT defined everywhere, but then this is appropriate
     if (!real_mmap)
     {
-        *(void **)(&real_mmap) = dlsym(RTLD_NEXT, "mmap");
+        *(void **)(&real_mmap) = dlsym(RTLD_NEXT, __func__);
```
Comment (Saviq) [inline tests/mir_test_framework/mmap_wrapper.cpp:60]: Will `__func__` ever be something else than `"mmap"`?

Comment (AlanGriffiths) [reply tests/mir_test_framework/mmap_wrapper.cpp:60]: If some sneaky implementation does `#define mmap mmap64`...

**Review by RAOF — APPROVED** [review]
Ah, yeah. That's a neat trick.

**Pull request merged**